pyteal/ast/asset.py

```diff
@@ -48,9 +50,9 @@ def total(cls, asset: Expr) -> MaybeValue:
             asset: An index into Txn.ForeignAssets that corresponds to the asset to check. Must
                 evaluate to uint64.
         """
-        require_type(asset.type_of(), TealType.uint64)
+        require_type(asset.type_of(), TealType.anytype)
```

Comment:
This should not change because the type remains a uint64, it just can now refer to the asset ID instead of its index in ForeignAssets.

Reply:
`AssetParam` type checking fixed.